extension/src/services/AvailableProjectsService.ts

```diff
@@ -0,0 +1,235 @@
+                } catch (error) {
+                    extensionLogOutputChannel.error(`Error initializing app hosts for new workspace folder ${addedFolder.name}: ${error}`);
+                    this._projects.set(addedFolder, null);
+                }
+            }
+        });
+
+        this._disposables.push(workspaceFoldersChangeListener);
+    }
+
+    private updateCandidateFile(folder: vscode.WorkspaceFolder, filePath: string, eventType: 'create' | 'change' | 'delete'): void {
+        const candidateSet = this._projects.get(folder);
+        if (!candidateSet) {
+            // If we don't have a set yet, ignore the event
+            return;
+        }
+
+        // Filter out build artifacts
+        if (this.isBuildArtifact(filePath)) {
+            return;
+        }
+
+        switch (eventType) {
+            case 'create':
+            case 'change':
+                candidateSet.add(filePath);
+                extensionLogOutputChannel.info(`Added/updated candidate: ${filePath} for ${folder.name}`);
+                break;
+            case 'delete':
+                candidateSet.delete(filePath);
+                extensionLogOutputChannel.info(`Removed candidate: ${filePath} for ${folder.name}`);
+                break;
+        }
+    }
+
+    private cleanupFileWatchersForFolder(folder: vscode.WorkspaceFolder): void {
+        // Get the watchers for this specific folder
+        const folderWatchers = this._folderWatchers.get(folder);
+        if (folderWatchers) {
+            // Dispose only the watchers for this folder
+            folderWatchers.forEach(watcher => watcher.dispose());
+
+            // Remove them from the main watchers array
+            this._fileWatchers = this._fileWatchers.filter(watcher => !folderWatchers.includes(watcher));
+
+            // Remove the folder from the tracking map
+            this._folderWatchers.delete(folder);
+        }
+    }
+
+    private async computeAppHostCandidates(folder: vscode.WorkspaceFolder): Promise<string[]> {
+        try {
+            // Find AppHost.cs and .csproj files in the given workspace folder
+            const appHostPattern = new vscode.RelativePattern(folder, '**/AppHost.cs');
+            const csprojPattern = new vscode.RelativePattern(folder, '**/*.csproj');
+
+            const [appHostUris, csprojUris] = await Promise.all([
+                vscode.workspace.findFiles(appHostPattern),
+                vscode.workspace.findFiles(csprojPattern)
+            ]);
+
+            const candidates = [
+                ...appHostUris.map(uri => uri.fsPath),
+                ...csprojUris.map(uri => uri.fsPath)
+            ];
+
+            // Filter out build artifacts and common excluded directories
+            const filteredCandidates = candidates.filter(filePath =>
+                !this.isBuildArtifact(filePath) && path.basename(filePath).includes('AppHost')
```

Comment:
Isn't there a property that tells us it's an AppHost?

Are there faster cross-plat ways to do file searches? File watchers are a headache.

Reply:
PS C:\dev\aspire-dogfood> findstr /S /I /M "IsAspireHost" *.cs*
Dogfood\Dogfood.AppHost\Dogfood.AppHost.csproj

Reply:
> How does the CLI do it today?

It actually runs msbuild on all of the csproj to determine whether IsAspireHost is true, and has incorrect logic for single-file apphost (AppHost.cs). I believe the maui extension does actually just check for a hardcoded string in the <UseMaui>  property, but the current approach is a temporary workaround anyway.